tests/unit/handlers/test_mssql.py

```diff
@@ -591,5 +593,265 @@ def test_types_casting(self):
+            driver="ODBC Driver 18 for SQL Server",
+            use_odbc=True,
+        )
+
+    def test_odbc_mode_enabled_with_driver_param(self):
+        """Test that ODBC mode is enabled when driver parameter is provided"""
+        handler = SqlServerHandler("mssql_odbc", connection_data=self.connection_data)
+        self.assertTrue(handler.use_odbc)
+
+    def test_odbc_mode_enabled_with_use_odbc_param(self):
+        """Test that ODBC mode is enabled when use_odbc parameter is True"""
+        connection_data = self.connection_data.copy()
+        del connection_data["driver"]
+        connection_data["use_odbc"] = True
+
+        handler = SqlServerHandler("mssql_odbc", connection_data=connection_data)
+        self.assertTrue(handler.use_odbc)
+
+    def test_odbc_mode_disabled_by_default(self):
+        """Test that ODBC mode is disabled when neither driver nor use_odbc is provided"""
+        connection_data = OrderedDict(
+            host="127.0.0.1",
+            port=1433,
+            user="example_user",
+            password="example_pass",
+            database="example_db",
+        )
+        handler = SqlServerHandler("mssql", connection_data=connection_data)
+        self.assertFalse(handler.use_odbc)
+
+    def test_odbc_connection_string_construction(self):
+        """Test that ODBC connection string is constructed correctly"""
+        mock_pyodbc = MagicMock()
+        mock_connect = MagicMock()
+        mock_pyodbc.connect = mock_connect
+
+        # Mock pyodbc in sys.modules so it can be imported
+        sys.modules["pyodbc"] = mock_pyodbc
+
+        try:
+            handler = SqlServerHandler("mssql_odbc", connection_data=self.connection_data)
+            handler.connect()
+        except Exception:
+            pass
+        finally:
+            if "pyodbc" in sys.modules:
+                del sys.modules["pyodbc"]
+        if mock_connect.called:
+            call_args = mock_connect.call_args
+            conn_str = call_args[0][0] if call_args[0] else ""
+
+            self.assertIn("DRIVER={ODBC Driver 18 for SQL Server}", conn_str)
+            self.assertIn("SERVER=127.0.0.1,1433", conn_str)
+            self.assertIn("DATABASE=example_db", conn_str)
+            self.assertIn("UID=example_user", conn_str)
+            self.assertIn("PWD=example_pass", conn_str)
+
+    def test_odbc_connection_with_encryption_params(self):
+        """Test that encryption parameters are added to connection string"""
+        mock_pyodbc = MagicMock()
+        mock_connect = MagicMock()
+        mock_pyodbc.connect = mock_connect
+
+        sys.modules["pyodbc"] = mock_pyodbc
+
+        connection_data = self.connection_data.copy()
+        connection_data["encrypt"] = "yes"
+        connection_data["trust_server_certificate"] = "yes"
+
+        try:
+            handler = SqlServerHandler("mssql_odbc", connection_data=connection_data)
+            handler.connect()
+        except Exception:
+            pass
```

Comment:
Can we get rid of this and the other occurrences of the same thing? This buries the actual issue when the test fails